src/components/Button/DefaultButton/index.tsx

```diff
@@ -0,0 +1,135 @@
+import { cn } from "@/utils";
+import { cva, VariantProps } from "class-variance-authority";
+import { ComponentProps, forwardRef } from "react";
+import { ButtonIconPosition, ButtonSize, ButtonVariant } from "../type";
+
+const buttonStyles = cva(
+  [
+    "w-full",
+    "rounded-md",
+    "font-semibold",
+    "focus:outline-none",
+    "disabled:cursor-not-allowed",
+    "disabled:text-[#72787F]",
+    "flex",
+    "items-center",
+    "justify-center"
+  ],
+  {
+    variants: {
+      variant: {
+        [ButtonVariant.DefaultPrimary]: "",
```

Comment:
???

Reply:
Sir it is used for giving button variants for different types of button , which I want to pick and I define these variants in enum.